lib/modules/datasource/docker/index.spec.ts

```diff
@@ -580,7 +615,7 @@ describe('modules/datasource/docker/index', () => {
       const tags = ['1.0.0'];
       httpMock
         .scope('https://registry.company.com/v2')
-        .get('/')
+        .get('/node/tags/list?n=10000')
```

Comment:
it seems we do the get request twice when no auch is required? this shouldn't happen and the request should be cached in memory.

🤔 @rarkins is request caching somehow disabled on tests?

Reply:
yes, so i thinking the cache should have catched the second call below 🤔

Reply:
This behavior was present before this MR. 
It is emphasize by using a specific path when getting the Auth header in order to fetch the tag list.

Before the MR
`.get('/')` is the call to get the header `getAuthHeaders`
`.get('/node/tags/list?n=10000')` is the call to get the tags  `getDockerApiTags`

The MR
`.get('/node/tags/list?n=10000')` is the call to get the header  `getAuthHeaders`
`.get('/node/tags/list?n=10000')` is the call to get the tags `getDockerApiTags`

Regarding the cache.
If I'm not missing some code, caching strategy is define at the method level not the remote api call level.

Reply:
Anything to fix change ?

Reply:
Ok, i see the issue, memcache isn't enabled. we should add a test with enabled cache

https://github.com/renovatebot/renovate/blob/637585c5fd9f7d3239318ad9e9e3ec288ec3c6e1/lib/util/http/index.ts#L149-L155

Reply:
hi @viceice,
 
Should I add memcache testing in this PR ? 
I feel  like testing this should be a full task in order to properly cover all accesses.

Reply:
We can defer cache test to a later pr.